fhir/com.b2international.snowowl.fhir.core/src/com/b2international/snowowl/fhir/core/provider/CodeSystemApiProvider.java

```diff
@@ -95,24 +105,27 @@ public boolean isSupported(String uri) {
 	}
 	
 	@Override
-	public CodeSystem getCodeSystem(LogicalId codeSystemLogicalId) {
+	public CodeSystem getCodeSystem(CodeSystemURI codeSystemURI) {
 		
-		String branchPath = codeSystemLogicalId.getBranchPath();
+		String branchPath = codeSystemURI.getPath();
 		
 		if (branchPath.equals(IBranchPath.MAIN_BRANCH)) {
-			throw FhirException.createFhirError(String.format("No code system version found for code system %s", codeSystemLogicalId), OperationOutcomeCode.MSG_PARAM_INVALID, "CodeSystem");
+			throw FhirException.createFhirError(String.format("No code system version found for code system %s", codeSystemURI.getUri()), OperationOutcomeCode.MSG_PARAM_INVALID, "CodeSystem");
 		} else {
+			
+			IBranchPath fullBranchPath = BranchPathUtils.createPath(BranchPathUtils.createMainPath(), branchPath);
```

Comment:
This won't work when the CodeSystem is an Extension. I think you don't actually need to special case the MAIN branch or compute any branch paths anymore. You need two API calls, one to fetch the CodeSystem (based on its ID from CodeSystemURI) and another to fetch one of its versions based on the version ID (which is the `path` part from the URI).